sycl/include/sycl/types.hpp

```diff
@@ -676,10 +676,26 @@ template <typename Type, int NumElements> class vec {
     using R = vec_data_t<convertT>;
     vec<convertT, NumElements> Result;
 
-    if constexpr (NativeVec && vec<convertT, NumElements>::NativeVec) {
+    // Whole vector conversion can only be done, if:
+    constexpr bool canUseNativeVectorConvert =
+#ifdef __NVPTX__
+        // - we are on CUDA, see intel/llvm#11840
```

Comment:
I understand the comment as "We are running on CUDA, see intel/llvm#11840 to understand why this is directly false.". Anyway, if it can lead to misunderstanding, we could clarify the comment to make sure it's understood by everyone in the same way.

Reply:
It should indeed contain `not`, because this line is a continuation of a previous statement, i.e. it should be read as:
```
Whole vector conversion can only be done, if:
- we are not on CUDA
- other conditions required for whole vector conversions
```

Added "not" in as part of merge commit (311ba0757eaeb7199bd9f7183eb8702cc70acfe4)